src/coreclr/jit/valuenum.cpp

```diff
@@ -14338,7 +14367,22 @@ bool Compiler::fgValueNumberHelperCall(GenTreeCall* call)
                 }
             }
 
+            if (isAlloc && ((call->gtCallMoreFlags & GTF_CALL_M_STACK_ARRAY) != 0))
```

Comment:
I suppose this could be changed to `FindWellKnownArg(StackArrayLocal) != nullptr` to get rid of the flag